rollout/controller.go

```diff
@@ -636,6 +636,7 @@ func (c *rolloutContext) getReferencedRolloutAnalyses() (*[]validation.AnalysisT
 			if err != nil {
 				return nil, err
 			}
+			templates.Args = blueGreen.PrePromotionAnalysis.Args
```

Comment:
1. Are you modifying the AnalysisTemplate which was pulled off the informer store (lister)? Objects received form the store should not be modified. See:

https://github.com/kubernetes/sample-controller/blob/b8d9e8c247129e53962d0dcfc08a4e8b47477318/controller.go#L322-L324

2. It seems we are clobbering the entire arguments of the template with the Arguments of the Rollout. Does this change support default values?

Reply:
> It seems we are clobbering the entire arguments of the template with the Arguments of the Rollout.

Can you elaborate on this? I thought we did this anyway when we created the AnalysisRun from the templates (i.e. combine the templates’ args with the AnalysisRunArguments passed through the RO spec)

Reply:
> Are you modifying the AnalysisTemplate which was pulled off the informer store (lister)?

I don't believe so. I'm copying the AnalysisRunArguments from the Rollout Spec into a `AnalysisTemplatesWithType` object, which is a struct I created for validation. I pass the templates and the args from the RO spec into `NewAnalysisRunFromTemplates` for validation, which doesn't seem to modify the templates.

Reply:
> Can you elaborate on this? 

Can you confirm that this is the behavior?

AnalysisTemplate
```yaml
args:
- name: aaa
  value: foo     # default value
- name: bbb  # mandatory supplied value
- name: ccc
  value: baz
```

Args supplied by rollout (note that aaa is omitted)
```yaml
args:
- name: bbb
  value: 11111
- name: ccc
  value: 2222
```

The AnalysisRun should be created with

```yaml
args:
- name: aaa
  value: foo
- name: bbb
  value: 11111
- name: ccc
  value: 2222
```

Reply:
> I don't believe so. I'm copying the AnalysisRunArguments from the Rollout Spec into a AnalysisTemplatesWithType object, which is a struct I created for validation.

Yes, but AnalysisTemplatesWithType contains a list of **_pointers to AnalysisTemplates and ClusterAnalysisTemplates_** which were pulled off the store, which you are directly modifying.

Reply:
> I don't believe so. I'm copying the AnalysisRunArguments from the Rollout Spec into a AnalysisTemplatesWithType object, which is a struct I created for validation

I took a closer look. You're right. I mistakenly thought it was returning AnalysisTemplates

Reply:
@jessesuen Can I merge this in then?

Reply:
My first concern is not valid, but can you answer my second concern?

Reply:
@jessesuen Yes, I confirm that the behavior in [this comment](https://github.com/argoproj/argo-rollouts/pull/1215#discussion_r639044608) is what happens.